Client/Frontend/Browser/Search/SearchEngines.swift

```diff
@@ -194,8 +194,8 @@ class SearchEngines {
   }
 
   func deleteCustomEngine(_ engine: OpenSearchEngine) throws {
-    // We can't delete a preinstalled engine or an engine that is currently the default.
-    if !engine.isCustomEngine || isEngineDefault(engine) {
+    // We can't delete a preinstalled engine
+    if !engine.isCustomEngine {
```

Comment:
I would probably use a guard here that way you know without even looking inside the block it will never execute further.
```
guard engine.isCustomEngine else { return }
```

Reply:
for context: we used to use `guard` only when some value is unwrapped, never for bool logic alone,
nowadays people use it as they want, i'm fine with both conventions

Reply:
I will keep as it is since we are also moving around the requirements here.